### PR TITLE
Fix/send form ui improvements

### LINKED ISSUE #2280: Send form: Visual improvements
- [x] **Incorrect size of the arrow between crypto <> fiat inputs**
Please check design. The icon is way too big now.

- [x] **Error / divider spacing**
Make the spacing for error between input and divider even
Screen Shot 2020-09-04 at 10 59 03

- [x] **Remove loading from the disabled primary button**
I believe there is no reason to have a loading in a disabled button (e.g. Changing RBF, Locktime, Broadcast results in jump to the right of the disabled primary button) 

- [x] **Consider some sort of scroll animation on "Add new recipient"** 
Adding an extra recipient doesn't provide strong enough action feedback (i.e. nothing "moves" besides the number of the recipient address header – relatively small "feedback" to the naked eye)

- [x] add space after "fee" text in "Total" row when sending eth token

### PR DESCRIPTION
**Pull request by slowbackspace**
close  https://github.com/trezor/trezor-suite/issues/2280

I couldn't figure out how to make framer motion work with react testing library (used in useSendForm test) so I disabled animations  in the test (it removes AnimatePresence and motion.div wrappers from the Output component).

### CODE REVIEW

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx [hunk 1]
```diff
@@ -35,27 +40,71 @@ const Row = styled.div`
...
+                                </Row>
+                            </>
+                        )}
+                    </OutputWrapper>
+                );
+
+                return disableAnim ? (
+                    outputComponent
+                ) : (
+                    <motion.div
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/send/components/Outputs/index.tsx:75]: you can still use only one nesting (OutputWrapper), just change to `styled(motion.div)` and disable animation for 1 output and tests as well

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx [hunk 2]
```diff
@@ -90,6 +90,9 @@ const Amount = ({ outputId }: { outputId: number }) => {
         composeTransaction,
     } = useSendFormContext();
 
+    // TODO: There is no guarantee that using arbitrary integer as an index won't result in undefined
+    if (!outputs[outputId]) return null;
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx]: interesting, is it caused by this animation? looks like a bug

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/views/wallet/send/components/Outputs/index.tsx:75]: i got an idea to move this animated wrapper to `/send/components` and use it in multiple places, like opening/closing additional options (locktime, data etc.)

**Review by szymonlesisz — CHANGES_REQUESTED** [review]
lets talk tomorrow how to make those animations usable in other parts

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx (hunk 2 above)
Comment (slowbackspace) [reply packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx]: my theory is that there is some delay (thanks to animation) before the Output component is removed from the DOM, so it technically still renders Amount component with index of an output that was already removed from reducer/context.

anyway imo it it still wrong that outputs[outputId] is not typed with potential `undefined`, if I pass out-of-range outputId TS will stay silent and app throws runtime error.

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx (hunk 1 above)
Comment (slowbackspace) [reply packages/suite/src/views/wallet/send/components/Outputs/index.tsx:75]: If I try to do OutputWrapper = styled(motion.div) I am getting a runtime error `Warning: React has detected a change in the order of Hooks called by ForwardRef(MotionComponent)....`, not sure what it doesn't like (maybe passing a ref?). At least I removed condition in return as it was not necessary.

Comment (slowbackspace) [reply packages/suite/src/views/wallet/send/components/Outputs/index.tsx:75]: I am all for adding animation to multiple places, but I would like this PR get merged before Friday as it addresses all requested improvements. Could we do that in another PR?

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx [hunk 3]
```diff
@@ -35,27 +40,62 @@ const Row = styled.div`
...
-                                <Address outputId={index} outputsCount={outputs.length} />
-                            </Row>
-                            <Row>
-                                <Amount outputId={index} />
-                            </Row>
-                        </>
-                    )}
-                </OutputWrapper>
+                <motion.div
+                    {...(outputs.length > 1 && !disableAnim ? ANIMATION.EXPAND : {})} // do not animate if there is only 1 output, prevents animation on clear
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/send/components/Outputs/index.tsx]: for cleaner code
create `const animation = outputs.length && !disableAnim ? ANIMATION.EXPAND : {};`
and use here:
`<motion.div {...animation}>`

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx [hunk 4]
```diff
@@ -35,27 +40,62 @@ const Row = styled.div`
...
+                    key={output.id}
+                    onAnimationComplete={onAddAnimationComplete}
+                >
+                    <OutputWrapper
+                        ref={index === outputs.length - 1 ? lastOutputRef : undefined} // set ref to last output
+                        index={index}
+                        key={output.id}
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/send/components/Outputs/index.tsx:76]: this key is not necessary anymore (set in motion.div)

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/views/wallet/send/components/Outputs/index.tsx [hunk 5]
```diff
@@ -35,27 +40,62 @@ const Row = styled.div`
...
+
+    useEffect(() => {
+        if (outputs.length < renderedOutputs) {
+            // updates rendered outputs count when removing an output
+            // this is necessary because onAddAnimationComplete is not fired when removing 2nd output
+            setRenderedOutputs(outputs.length);
+        }
+    }, [outputs.length, renderedOutputs, setRenderedOutputs]);
+
+    const WrapperComponent = disableAnim ? Wrapper : AnimatePresence;
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/send/components/Outputs/index.tsx]: is this necessary since you disable animations in lower block?

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

**Review by szymonlesisz — APPROVED** [review]
nice!

**Pull request merged**